xarray/core/missing.py

```diff
@@ -345,8 +346,13 @@ def interp_na(
                     f"Expected integer or floating point max_gap since use_coordinate=False. Received {max_type}."
                 )
 
-    # method
-    index = get_clean_interp_index(self, dim, use_coordinate=use_coordinate)
+    if use_coordinate:
+        index = self[dim]
+    else:
+        if xr.core.utils.is_monotonic_decreasing(self[dim]):
+            index = xr.IndexVariable(dim, np.sort(self[dim])[::-1])
+        else:
```

Comment:
We could also check `is_monotonic_increasing`, and raise an error is the index is not monotonic in either direction.

Reply:
> Hi @alrho007, thanks for submitting this.
> 
> It looks like you might have found the source of the problem! However there are a few things we would need to fix with this PR before it can be merged.
> 
> 1. **The tests are failing** You can see that multiple automated tests have failed on this pull request. A couple of these are possibly unrelated to your changes, but I think a lot of them were caused by your change. Did you run the tests locally first to see if they passed?
> 2. **You need to add your own test** Pull requests altering the code require you to write at least one new test to check that your change fixes the behaviour you think it fixes. In your case that means you need to write a test that creates a dataarray with a monotonically decreasing index, calls `interp_na` on it, then checks that the result is as you expected. This test should fail without your changes to the code, but pass with your changes.
> 3. **You've deleted two files** (`doc/whats-new.rst` and `doc/api.rst`) You appear to have deleted two files from the codebase, and then committed those changes. If I merge this pull request as it is it would delete those files from xarray.
> 4. **Your branch name** I notice the name of this branch is `round-stringify`, which seems to be the same as one of your earlier pull requests. What you should do is create different branches for different pull requests, to keep unrelated changes separate. How to create a new branch is explained in the contributing guide. I would suggest creating a new branch named `monotonic_decreasing` or similar with just the relevant changes.
> 
> If you can address these things then we can merge your changes!

Thank you very much, I'll work on them right away. I think I messed up some commits when I was rushing because of my  final application.